src/test/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperatorTest.scala

```diff
@@ -37,16 +37,30 @@ class TezosNodeOperatorTest extends FlatSpec with MockFactory with Matchers with
     (tezosRPCInterface.runAsyncGetQuery _).when("zeronet", "blocks/head~").returns(Future.successful(TezosResponseBuilder.blockResponse))
     (tezosRPCInterface.runAsyncGetQuery _).when("zeronet", "blocks/head/operations").returns(Future.successful(TezosResponseBuilder.operationsResponse))
 
-    (tezosRPCInterface.runBatchedGetQuery[Int] _).when("zeronet", (0 to 162099).toList, *, *).returns(Future.successful(List((0, TezosResponseBuilder.batchedGetQueryResponse))))
-    (tezosRPCInterface.runBatchedGetQuery[BlockHash] _).when("zeronet", *, *, *).returns(Future.successful(List((BlockHash("BMKoXSqeytk6NU3pdL7q8GLN8TT7kcodU1T6AUxeiGqz2gffmEF"), TezosResponseBuilder.batchedGetQuerySecondCallResponse))))
+
+    (tezosRPCInterface.runBatchedGetQuery[Int] _)
+      .when("zeronet", *, *, *)
+      .returns(Future.successful(List((0, TezosResponseBuilder.batchedGetQueryResponse))))
+      .anyNumberOfTimes
+
+    (tezosRPCInterface.runBatchedGetQuery[BlockHash] _)
+      .when("zeronet", *, *, *)
+      .returns(Future.successful(List((BlockHash("BMKoXSqeytk6NU3pdL7q8GLN8TT7kcodU1T6AUxeiGqz2gffmEF"), TezosResponseBuilder.batchedGetQuerySecondCallResponse))))
+      .anyNumberOfTimes
 
     val nodeOp: TezosNodeOperator = new TezosNodeOperator(tezosRPCInterface, config)
 
     //when
-    val block: Future[List[(TezosTypes.Block, List[TezosTypes.AccountId])]] = nodeOp.getAllBlocks("zeronet")
+    val blockPages: Future[nodeOp.PaginatedBlocksResults] = nodeOp.getLatestBlocks("zeronet")
 
     //then
-    block.futureValue should have length 1
+    val (pages, total) = blockPages.futureValue
+    total shouldBe 162100
+    val results = pages.toList
+    results should have length 163
+    logger.info("First page is {}", results.head.futureValue)
+    results.head.futureValue should have length 1
```

Comment:
I have some issue to figure out an unexpected error during this test.
The error is the same found by the travis CI integration.

Any help is appreciated.

Reply:
Indeed, I agree, I made all sorts of mix-and-try this morning to see what combination could work, yet to no avail.

I have not enough expertise on how to use advanced scalamock features, or (as I fear) the framework can't possibly express the needed behaviour in a simple way

Reply:
I've created basic example of a failing test, check if it represents what you wanted to so I could create an issue in the scalamock repo:
```
  class Sth {
    def aaa[A](i: Int): A = ???

    def bbb = aaa[Int](0)
    def ccc = aaa[String](0)
  }

  "X" should "Y" in {
    val sth = stub[Sth]
    (sth.aaa[Int] _).when(*).returns(1)
    (sth.aaa[String] _).when(*).returns("a")

    sth.bbb shouldBe 1
    sth.ccc shouldBe "a"
  }
```

Reply:
> I've created basic example of a failing test, check if it represents what you wanted to so I could create an issue in the scalamock repo:
> 
> ```
>   class Sth {
>     def aaa[A](i: Int): A = ???
> 
>     def bbb = aaa[Int](0)
>     def ccc = aaa[String](0)
>   }
> 
>   "X" should "Y" in {
>     val sth = stub[Sth]
>     (sth.aaa[Int] _).when(*).returns(1)
>     (sth.aaa[String] _).when(*).returns("a")
> 
>     sth.bbb shouldBe 1
>     sth.ccc shouldBe "a"
>     sth.bbb shouldBe 1
>     sth.ccc shouldBe "a"
>   }
> ```

is more like what we need

Thanks